### ISSUE TITLE
Fine-tune job record never updates status/phase/step/loss (metrics path works)

### ISSUE THREAD

**Issue by kreuzhofer** [kreuzhofer/dgx-manager#81]
## Problem

`/api/finetune/:id` reports `status: "starting"`, `phase: null`, `step: null`, `loss: null` for the **entire life of a training run** — including after it has completed hundreds of steps. Meanwhile `/api/finetune/:id/metrics` has every row, correct.

So the metrics path works and the **job-record update path does not**. The dashboard shows a frozen header above a live, correct loss chart.

## Measured 2026-09-03/04

Job `cmtm0qe244qwm2auh9rdr28mv` (5-step smoke) and `cmtmt9dbd57gi2auhok8cptou` (500-step), both on dgx-spark-04:

```
/api/finetune/:id          → status=starting  phase=None  step=None  loss=None
/api/finetune/:id/metrics  → [{"step":1,"loss":3.741,...}, … {"step":500,"loss":0.400}]
```

The 500-step job stayed `starting` through all 500 steps **and** its ~30-minute end-of-run eval, only flipping to `completed` at the very end. Training was demonstrably running throughout (container up, GPU 85-96%, `[TRAIN] step=N/500` lines in `train.log`, metrics rows arriving).

## Impact

- The dashboard is unusable for judging whether a run is progressing — the one place an operator looks first.
- It cost real diagnostic time here: a job that was healthy looked stalled, and distinguishing "silent because it is initialising" from "silent because it is wedged" required SSHing to the node and reading `train.log` and `ps` directly.
- It defeats the purpose of the phase tracking listed as shipped under Phase 3 in `docs/ROADMAP.md` ("container → downloading → loading → tokenizing → training → eval → saving").

## Where to look

The agent emits both on the same callback surface (`packages/agent/src/index.ts` `cmd:finetune:start` handler): `onProgress` sends `agent:finetune:progress` with `phase`, `step`, `loss`, `lr`, `evalLoss`. Metrics rows clearly arrive, so the message is being sent and received — the divergence is likely in how the server persists that message to the `FineTuneJob` row versus to `MetricSnapshot`/the metrics table.

Note `phase` is `null` rather than stale, suggesting it is never written at all rather than written once and not updated.

## Suggested test

Integration test asserting that after N progress messages the job row reflects the latest `step`/`loss`/`phase`, not just that metrics rows exist. The current tests evidently cover the metrics path only.

Found while working [#79](https://github.com/kreuzhofer/dgx-manager/issues/79).
